src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreator.java

```diff
@@ -228,6 +235,18 @@ public void updateInputRequestWithLatestSidamUserInfo(UserProfileCreationData pr
         }
     }
 
+    public void updateSidamUserInfoWithUserProfileDetails(UserProfileCreationData profileData,
+                                                          IdamRolesInfo idamRolesInfo) {
+        profileData.setStatus(IdamStatusResolver.resolveIdamStatus(idamRolesInfo));
+        if (!(profileData.getFirstName().equals(idamRolesInfo.getForename()) && profileData.getLastName()
+                .equals(idamRolesInfo.getSurname()))) {
+            UpdateUserDetails updateUserDetails = new UpdateUserDetails(profileData.getFirstName(),
+                    profileData.getLastName(), "ACTIVE".equals(profileData.getStatus().toString()));
+            idamService.updateUserDetails(updateUserDetails, idamRolesInfo.getId());
```

Comment:
Sorry @sahitya-desireddy - just one thought - you need to handle the scenario where idam returns non 200 response then you need to audit the exception. @shreedhar-hmcts correct me if I am wrong here

Reply:
Also if idamService.updateUserDetails fails then how the error response will be returned to SRD ?